home-assistant-plugin/custom_components/xknx/climate.py

```diff
@@ -93,6 +96,9 @@
         vol.Required(CONF_TEMPERATURE_ADDRESS): cv.string,
         vol.Required(CONF_TARGET_TEMPERATURE_STATE_ADDRESS): cv.string,
         vol.Optional(CONF_TARGET_TEMPERATURE_ADDRESS): cv.string,
+        vol.Optional(CONF_SETPOINT_SHIFT_MODE, default=DEFAULT_SETPOINT_SHIFT_MODE): cv.enum(
+            SetpointShiftMode
+        ),
```

Comment:
Again HA has very strict styling guidelines. These shall be ordered the following way: Optionals with defaults > Required > Optionals without defaults. So I'd move it to line 87.